server/sonar-process-monitor/src/main/java/org/sonar/application/SchedulerImpl.java

```diff
@@ -223,8 +222,16 @@ public void onAppStateOperational(ProcessId processId) {
 
   @Override
   public void onProcessState(ProcessId processId, Lifecycle.State to) {
-    if (to == Lifecycle.State.STOPPED) {
-      onProcessStop(processId);
+    switch (to) {
+      case STOPPED:
+        onProcessStop(processId);
+        break;
+      case STARTING:
+        stopCountDown.incrementAndGet();
```

Comment:
my main concern here is that if someone is launching a process without this function, the automatic won't work. Since the stop is based on listener (decreasing the countdown), seems logical to do the same for starting a process.